extractors/building_parser.py

```
@@ -310,202 +309,142 @@ def extract_image_based_pdf_improved(self, doc, last_word: str) -> Optional[Dict
+            print("면적 보정 시도: 라벨-숫자 순서 기반 매핑")
+            block_result = self.parse_area_block(lines)
+
+            # parse_area_block 결과가 있으면 덮어씀
+            if "대지면적" in block_result:
+                info["대지면적"] = block_result["대지면적"]
+            if "연면적" in block_result:
+                info["연면적"] = block_result["연면적"]
+
+            # 상식 검증: 연면적 < 대지면적이면 스왑 (숫자 두 개만 있을 때 오인식 방지)
+            if info.get("대지면적") and info.get("연면적"):
+                if info["연면적"] < info["대지면적"]:
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)